### ISSUE TITLE
I think there is a BUG in the eval code.

### ISSUE THREAD

**Issue by sean-wade** [ultralytics/yolov5#8168]
https://github.com/ultralytics/yolov5/blob/6adc53ba5fd7e87002ef80603f3fcf1c6ace72f5/val.py#L230

If there's no predictions, line 230 will continue the loop and put an empty torch.zeros((3, 0)) into stats. 
But the gt-label numbers will be less accumulated.

I think it should be like this one: 

stats.append((correct, *torch.zeros((2, 0), device=device), labels[:, 0]))

**Comment by glenn-jocher** [comment]
@sean-wade hi, thank you for your feature suggestion on how to improve YOLOv5 🚀! I think in theory this may have an effect but in practice val.py will always generate predictions as the confidence threshold is set to 0.001 generating thousands of FPs that are collected at the bottom right of the PR curve. 

The fastest and easiest way to incorporate your ideas into the official codebase is to submit a Pull Request (PR) implementing your idea, and if applicable providing before and after profiling/inference/training results to help us understand the improvement your feature provides. This allows us to directly see the changes in the code and to understand how they affect workflows and performance.

Please see our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/) to get started.

**Comment by sean-wade** [comment]
Yes, the probability of no predictions is quite low.

**Comment by glenn-jocher** [comment]
@sean-wade even so we still might benefit from a PR to update this. Can you please submit a PR with before and after val results with your proposed fix? Thanks!
